app.rb

```diff
@@ -0,0 +1,200 @@
+
+#
+# Kill Bill configuration and helpers
+#
+
+KillBillClient.url = settings.kb_url
+
+# Multi-tenancy and RBAC credentials
+options = {
+    :username => 'admin',
+    :password => 'password',
+    :api_key => 'bob',
+    :api_secret => 'lazar'
+}
+
+# Audit log data
+user = 'demo'
+reason = 'New subscription'
+comment = 'Trigger by Sinatra'
+
+def create_kb_account(user, reason, comment, options)
+  account = KillBillClient::Model::Account.new
+  account.name = 'John Doe'
+  account.currency = 'USD'
+  account.create(user, reason, comment, options)
+end
+
+def create_kb_payment_method(account, funding_source, customer_id, user, reason, comment, options)
+  puts "Customer ID: #{customer_id}"
+  puts "Funding Source: #{funding_source}"
+
+  pm = KillBillClient::Model::PaymentMethod.new
+  pm.account_id = account.account_id
+  pm.plugin_name = 'killbill-dwolla'
+  pm.plugin_info = {'fundingSource' => funding_source, 'customerId' => customer_id }
+  pm.create(true, user, reason, comment, options)
+end
+
+def create_subscription(account, user, reason, comment, options)
+  subscription = KillBillClient::Model::Subscription.new
+  subscription.account_id = account.account_id
+  subscription.product_name = 'Sports'
+  subscription.product_category = 'BASE'
+  subscription.billing_period = 'MONTHLY'
+  subscription.price_list = 'DEFAULT'
+  subscription.price_overrides = []
+
+  # For the demo to be interesting, override the trial price to be non-zero so we trigger a charge in Stripe
+  override_trial = KillBillClient::Model::PhasePriceOverrideAttributes.new
+  override_trial.phase_type = 'TRIAL'
+  override_trial.fixed_price = 10.0
+  subscription.price_overrides << override_trial
+
+  subscription.create(user, reason, comment, nil, true, options)
+end
+
+#
+# Sinatra handlers
+#
+
+get '/' do
+
+  puts "--------------------------------------------------"
+  puts "client_id = #{settings.client_id}"
+  puts "client_secret = #{settings.client_secret}"
+  puts "access_token = #{settings.dwolla_access_token}"
+  puts "--------------------------------------------------"
+
+  # see dwolla.com/applications for your client id and secret
+  $dwolla = DwollaV2::Client.new(id: settings.client_id,
+                                 secret: settings.client_secret
+  ) do |config|
+    config.environment = :sandbox
+  end
+
+  # generate a token on dwolla.com/applications
+  account_token = $dwolla.tokens.new access_token: settings.dwolla_access_token
+
+  # Using DwollaV2 - https://github.com/Dwolla/dwolla-v2-ruby (Recommended)
+  customers = account_token.get "customers", search: "no@nemail.net"
+
+  if !customers.nil? and customers._embedded.customers.size > 0
+    customer = customers._embedded.customers[0]
+    customer_url = customer._links.self.href
+    puts "Customer found: #{customer_url}"
+  else
+    request_body = {
+        :firstName => 'Jane',
+        :lastName => 'Merchant',
+        :email => 'no@nemail.net',
```

Comment:
Should probably be in sync with the KB account (you could keep the name hardcoded and generate one-off emails based on timestamp).